### PR TITLE
Add the dirty-ring of vm features

### PR DESCRIPTION
**Pull request by iccaszhulili**
Signed-off-by: Lili Zhu <lizhu@redhat.com>

### CODE REVIEW

**Comment by iccaszhulili** [comment]
(3/3) type_specific.io-github-autotest-libvirt.virsh.domdirtyrate_calc.positive.dirty_ring_mode: STARTED
 (3/3) type_specific.io-github-autotest-libvirt.virsh.domdirtyrate_calc.positive.dirty_ring_mode: PASS (90.31 s)

File: virttest/libvirt_xml/vm_xml.py [hunk 1]
```diff
@@ -3223,7 +3223,8 @@ class VMFeaturesXML(base.LibvirtXMLBase):
                  'hyperv_reenlightenment_state', 'hyperv_vpindex_state',
                  'kvm_hidden_state', 'pvspinlock_state', 'smm', 'hpt',
                  'htm', 'smm_tseg_unit', 'smm_tseg', 'nested_hv',
-                 'pmu', 'kvm_poll_control', 'ioapic')
+                 'pmu', 'kvm_poll_control', 'ioapic',
+                 'kvm_dirty_ring_state', 'kvm_dirty_ring_size')
```
Comment (chunfuwen) [inline virttest/libvirt_xml/vm_xml.py:3227]: please update doc string after line 3218 to match attributes change

Comment (iccaszhulili) [reply virttest/libvirt_xml/vm_xml.py:3228]: Updated. Thanks

**Review by chunfuwen — APPROVED** [review]
LGTM

File: virttest/libvirt_xml/vm_xml.py [hunk 2]
```diff
@@ -3319,6 +3321,16 @@ def __init__(self, virsh_instance=base.virsh):
                                  libvirtxml=self,
                                  parent_xpath='/',
                                  tag_name='ioapic')
+        accessors.XMLAttribute(property_name='kvm_dirty_ring_state',
```
Comment (dzhengfy) [inline virttest/libvirt_xml/vm_xml.py:3324]: The best way to to create a new XML class as nested one under <features>, for example, VMFeatureKvmXML for below xml. There is an enhancement we can do, but maybe in another PR. For current PR, it is fine.  
```
<features>
...
  <kvm>
    <hidden state='on'/>
    <hint-dedicated state='on'/>
    <poll-control state='on'/>
    <pv-ipi state='off'/>
    <dirty-ring state='on' size='4096'/>
  </kvm>
...
</features>
```

**Review by dzhengfy — APPROVED** [review]

File: virttest/libvirt_xml/vm_xml.py (hunk 2 above)
Comment (dzhengfy) [reply virttest/libvirt_xml/vm_xml.py:3324]: @chloerh , FYI.

Comment (chloerh) [reply virttest/libvirt_xml/vm_xml.py:3324]: @dzhengfy Agreed.

**Comment by dzhengfy** [comment]
@chloerh is in PTO now, but I need merge this pr now considering chloerh already gave confirmation.

**Pull request merged**